specs/DisableCrashReporting.md

```diff
@@ -0,0 +1,120 @@
+Custom Crash Reports
+===
+
+# Background
+Currently, when a WebView2 is created a WebView2 crashpad process will be created in addition to starting a WebView2 browser process. If any WebView2 process crashes, the crashpad process will create crash dumps and send them to Microsoft for diagnosis. This document covers new APIs to allow the end developer to customize crash reporting to help when running diagnostics and doing analysis. They can set the `CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled` property to not have crash dumps sent to Microsoft and use the `CoreWebView2Environment.CrashDumpFolderPath` property to locate crash dumps and do customization with them instead.
+
+# Examples
+## WinRT and .NET   
+```c#
+
+/// Create WebView Environment with option
+void CreateEnvrionmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.IsCustomCrashReportingEnabled = true;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    webView.CoreWebView2.ProcessFailed += WebView_ProcessFailed;
+}
+
+void WebView_ProcessFailed(object sender, CoreWebView2ProcessFailedEventArgs e)
+{
+    // When process failed, do custom parsing with dumps
+    string crashDumpFolder = webView.CoreWebView2.Environment.CrashDumpFolderPath;
+    ProcessNewCrashDumps(crashDumpFolder);
```

Comment:
This implies that the ProcessFailed event is not raised as soon as a process fails, but rather waits for the crash dump to be created.

But an app may not want to wait for the crash dump to be created. As soon as the render process crashes, it wants to quickly spin up another one, so the user sees the app recover immediately, rather than staring at a black screen for 15 seconds while the crash dump is created. Is there an event that gets raised before we start generating the crash dump?

Reply:
This is a sample for developer who want to do custom parsing with the dump. ProcessFailed is a good place for those people. Otherwise, we are not holding on sending process failed notification when process fails

Reply:
Please further investigate that this isn't just a race that happened to go the way we wanted one time. 
Please also update sample to use the ProcessFailed event to poke the file system watcher (and add a file system watcher)

Reply:
Updated with new sample code